docs/docs/plugin.md

```diff
@@ -0,0 +1,169 @@
+than the configuration from the plugin descriptor is passed to every created instance of that plugin, before any other method is called.
+
+
+## Plugin Descriptor
+
+You can extend Marathon with a set of plugins. 
+To make sure only valid combinations of plugins are taken, you have to provide a plugin descriptor.
+That descriptor defines which plugin interface is extended by which service provider.
+Marathon will load this descriptor with all defined plugins during start up.
+ 
+Example:
+
+```json
+{
+  "plugins": [
+    {
+      "plugin": "mesosphere.marathon.plugin.auth.Authorizer",
+      "implementation": "my.company.ExampleAuthorizer"
+    },
+    {
+      "plugin": "mesosphere.marathon.plugin.auth.Authenticator",
+      "implementation": "my.company.ExampleAuthenticator",
+      "configuration": {
+        "users": [
+          {
+            "user": "ernie",
+            "password" : "ernie",
+            "permissions": [
+              { "allowed": "create", "on": "/dev/" }
+            ]
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+The configuration section of one plugin is optional. If provided, the json object is given as is to the plugin.
+
+## Start Marathon with plugins
+
+In order to make your plugins available to Marathon you have to
+
+- place your plugin.jar with all dependant jars (that are not already provided by Marathon) into a directory
+- write a plugin descriptor, that list the plugins to use
+
+Now you can start Marathon with this options:
+
+- `--plugin_dir path` where path is /the/directory/with/your/plugin.jar
+- `--plugin_conf conf.json` where conf.json is /path/to/plugin-descriptor.json
+
+
+# Available Plugin Extensions
+
+## Security
+
+#### [mesosphere.marathon.plugin.auth.Authenticator](https://github.com/mesosphere/marathon/blob/master/plugin-interface/src/main/scala/mesosphere/marathon/plugin/auth/Authenticator.scala)
+
+Marathon exposes an HTTP REST API. To make sure only authenticated identities can access the system, the plugin developer has to implement the Authenticator interface.
+This interface relies purely on the HTTP Layer. Please see the [Authenticator](https://github.com/mesosphere/marathon/blob/master/plugin-interface/src/main/scala/mesosphere/marathon/plugin/auth/Authenticator.scala) trait for documentation, as well as the [Example Scala Plugin](https://github.com/mesosphere/marathon-example-plugins/tree/master/auth) or the [Example Java Plugin](https://github.com/mesosphere/marathon-example-plugins/tree/master/javaauth)
+
+#### [mesosphere.marathon.plugin.auth.Authorizer](https://github.com/mesosphere/marathon/blob/master/plugin-interface/src/main/scala/mesosphere/marathon/plugin/auth/Authorizer.scala)
+
+This plugin plays along the Authentication plugin. With this interface you can refine, which authenticated identities can access which resource with which action. 
```

Comment:
s/With this interface you can refine, which authenticated identities can access which resource with which action./With this interface you can refine, what actions an authenticated identity can perform on Marathon's resources/